staging/src/k8s.io/kubectl/pkg/cmd/apply/patcher.go

```diff
@@ -118,7 +122,17 @@ func (p *Patcher) patchSimple(obj runtime.Object, modified []byte, namespace, na
 	var patchType types.PatchType
 	var patch []byte
 
-	if p.OpenapiSchema != nil {
+	if p.OpenAPIV3Root != nil {
+		// TODO: Capture error and log maybe?
+		if gvkSupported, _ := p.gvkSupportsPatchOpenAPIV3(p.Mapping.GroupVersionKind); gvkSupported {
+			patch, err = p.buildStrategicMergePatchFromOpenAPIV3(original, modified, current)
```

Comment:
it seems like these two things are running together yet they have a lot of common code? Can we actually return the `PatchMeta` if we find one?

Reply:
I like keeping the two separate because they look at different components of the OpenAPI. `gvkSupportsPatchOpenAPIV3` looks at the PATCH verb for a resource path (does not look at schema at all) while `buildStrategicMergePatchFromOpenAPIV3` looks at Component definitions and traverses the schema resource.